staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -6104,3 +6119,42 @@ const (
 	// and data streams for a single forwarded connection
 	PortForwardRequestIDHeader = "requestID"
 )
+
+// PortStatus represents the condition of a service port condition
+type PortStatus struct {
+	// The port that will be exposed by this service.
+	Port int32 `json:"port" protobuf:"varint,1,opt,name=port"`
+	// The IP protocol for this port. Supports "TCP", "UDP", and "SCTP".
+	// Default is TCP.
+	// +optional
```

Comment:
If this is optional with a default, you need to write the rule in defaults.go to set it.